### ISSUE TITLE
"lottie-ios.modulemap not found" regression with CocoaPods 1.7.0 beta

### ISSUE THREAD

**Issue by chrisballinger** [airbnb/lottie-ios#859]
## Check these before submitting:
- [x] The issue doesn't involve an [Unsupported Feature](https://github.com/airbnb/lottie-ios/blob/master/README.md#supported-after-effects-features)
- [x] This issue isn't related to another open issue

## This issue is a:
- [] Non-Crashing Bug (Visual or otherwise)
- [] Crashing Bug
- [] Feature Request
- [x] Regression (Something that once worked, but doesn't work anymore)

## Which Version of Lottie are you using?

Lottie 3.0

## What Platform are you on? 

- [] MacOS
- [x] iOS

## What Language are you in?
- [x] Swift
- [] Objective-C

Although this is not necessarily a bug in Lottie itself, I wanted to preemptively warn the team that there is a regression in CocoaPods 1.7.0 beta that prevents the usage of Lottie with static linking. The upstream issue in CocoaPods is here: https://github.com/CocoaPods/CocoaPods/issues/8695

An example project exhibiting this issue is available here in the `1.7.0` folder: https://github.com/chrisballinger/cocoapods-issue-8695

You will see the following issue when compiling:

```
<unknown>:0: error: module map file '/Pods/Headers/Public/Lottie/lottie-ios.modulemap' not found
<unknown>:0: error: underlying Objective-C module 'Lottie' not found
```

My initial assumption that the lottie-ios.podspec module name override was associated with the problem, but the issue persists even without the module name override. Will update this issue when I uncover more information.

Possible workarounds include:
* `use_frameworks!` - this only seems to affect integration as a static library
* Use CocoaPods 1.6.1 or lower
* Use an alternate integration method like Carthage

**Comment by chrisballinger** [comment]
The issue seems to be the dash in the pod name. Changing the podspec name to `Lottie` from `lottie-ios` to the following resolves the issue:

# Works
```ruby
Pod::Spec.new do |s|
  s.name             = 'Lottie'
  s.version          = '3.0.4'
  s.summary          = 'A library to render native animations from bodymovin json. Now in Swift!'

s.description      = <<-DESC
Lottie is a mobile library for Android and iOS that parses Adobe After Effects animations exported as json with bodymovin and renders the vector animations natively on mobile and through React Native!

For the first time, designers can create and ship beautiful animations without an engineer painstakingly recreating it be hand. Since the animation is backed by JSON they are extremely small in size but can be large in complexity! Animations can be played, resized, looped, sped up, slowed down, and even interactively scrubbed.
DESC

  s.homepage         = 'https://github.com/airbnb/lottie-ios'
  s.license          = { :type => 'Apache', :file => 'LICENSE' }
  s.author           = { 'Brandon Withrow' => 'buba447@gmail.com' }
  s.source           = { :git => 'https://github.com/airbnb/lottie-ios.git', :tag => s.version.to_s }
  
  s.swift_version = '4.2'
  s.ios.deployment_target = '9.0'
  s.osx.deployment_target = '10.10'
  s.tvos.deployment_target = '9.0'
  
  s.source_files = 'lottie-swift/src/**/*'
  s.ios.source_files   = 'lottie-swift/iOS/*.swift'
  s.ios.exclude_files   = 'lottie-swift/src/Public/MacOS/**/*'
  s.tvos.exclude_files   = 'lottie-swift/src/Public/MacOS/**/*'
  s.osx.exclude_files = 'lottie-swift/src/Public/iOS/**/*'
  
  s.ios.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.tvos.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.osx.frameworks = ['AppKit', 'CoreGraphics', 'QuartzCore']
  s.header_dir = 'Lottie'
end
```

# Doesn't work

```ruby
Pod::Spec.new do |s|
  s.name             = 'lottie-ios'
  s.version          = '3.0.4'
  s.summary          = 'A library to render native animations from bodymovin json. Now in Swift!'

s.description      = <<-DESC
Lottie is a mobile library for Android and iOS that parses Adobe After Effects animations exported as json with bodymovin and renders the vector animations natively on mobile and through React Native!
For the first time, designers can create and ship beautiful animations without an engineer painstakingly recreating it be hand. Since the animation is backed by JSON they are extremely small in size but can be large in complexity! Animations can be played, resized, looped, sped up, slowed down, and even interactively scrubbed.
DESC

  s.homepage         = 'https://github.com/airbnb/lottie-ios'
  s.license          = { :type => 'Apache', :file => 'LICENSE' }
  s.author           = { 'Brandon Withrow' => 'buba447@gmail.com' }
  s.source           = { :git => 'https://github.com/airbnb/lottie-ios.git', :tag => s.version.to_s }
  
  s.swift_version = '4.2'
  s.ios.deployment_target = '9.0'
  s.osx.deployment_target = '10.10'
  s.tvos.deployment_target = '9.0'
  
  s.source_files = 'lottie-swift/src/**/*'
  s.ios.source_files   = 'lottie-swift/iOS/*.swift'
  s.ios.exclude_files   = 'lottie-swift/src/Public/MacOS/**/*'
  s.tvos.exclude_files   = 'lottie-swift/src/Public/MacOS/**/*'
  s.osx.exclude_files = 'lottie-swift/src/Public/iOS/**/*'
  
  s.ios.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.tvos.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.osx.frameworks = ['AppKit', 'CoreGraphics', 'QuartzCore']
  s.module_name = 'Lottie'
  s.header_dir = 'Lottie'
end
```

**Comment by Coeur** [comment]
It's unlikely that we'll rename the pod, so please report the regression to https://github.com/cocoapods/cocoapods/issues instead.

**Comment by chrisballinger** [comment]
Aye! This issue should be closed once 1.7.0beta4 is released

**Comment by chrisballinger** [comment]
Resolved as of CocoaPods [1.7.0.rc.1](https://github.com/CocoaPods/CocoaPods/releases/tag/1.7.0.rc.1)

**Comment by pluckq** [comment]
Reproduces with AsyncDisplayKit/Texture framework
